packages/libs/s3-static-assets/src/index.ts

```diff
@@ -24,149 +23,160 @@ type UploadStaticAssetsOptions = {
   publicDirectoryCache?: PublicDirectoryCache;
 };
 
+type AssetDirectoryFileCachePoliciesOptions = {
+  basePath: string;
+  // The directory containing the build output.
+  // .i.e. by default .serverless_nextjs
+  serverlessBuildOutDir: string;
+  nextStaticDir?: string;
+  publicDirectoryCache?: PublicDirectoryCache;
+};
+
 /**
- * Uploads from built assets folder in .serverless_nextjs/assets to S3.
- * This is used to decouple a build from deployment.
- * Currently this works for Lambda@Edge deployment.
+ * Returns an array of files with with their relevant cache policies.
  */
-const uploadStaticAssetsFromBuild = async (
-  options: UploadStaticAssetsOptions
-): Promise<AWS.S3.ManagedUpload.SendData[]> => {
-  const {
-    bucketName,
-    credentials,
-    basePath,
-    publicDirectoryCache,
-    nextConfigDir
-  } = options;
-  const s3 = await S3ClientFactory({
-    bucketName,
-    credentials: credentials
-  });
+const getAssetDirectoryFileCachePolicies = (
```

Comment:
I've taken all the logic from `uploadStaticAssetsFromBuild` that determines the cache-control for each file in `/assets` and refactored it to be agnostic to the deployment mechanism. This file is imported into the CDK construct now, whilst also used below in `uploadStaticAssetsFromBuild`. I think we could do with a grander refactor of this form that removes the deployment code from the synthesis.

Reply:
This didn't make sense to use in the end as the data structure required to work with the constructs didn't fit this output enough to make it worth it. Happy to revert the changes to this module if you think its needed.

Reply:
I think it should be ok to keep it.